jest.config.ts

```diff
@@ -0,0 +1,58 @@
+import type { Config } from '@jest/types';
+import { version as typescriptESLintPluginVersion } from '@typescript-eslint/eslint-plugin/package.json';
+import { version as eslintVersion } from 'eslint/package.json';
+import * as semver from 'semver';
+
+const config = {
+  clearMocks: true,
+  restoreMocks: true,
+  resetMocks: true,
+
+  coverageThreshold: {
+    global: {
+      branches: 100,
+      functions: 100,
+      lines: 100,
+      statements: 100,
+    },
+  },
+
+  projects: [
+    {
+      displayName: 'test',
+      testPathIgnorePatterns: [
+        '<rootDir>/lib/.*',
+        '<rootDir>/src/rules/__tests__/fixtures/*',
+        '<rootDir>/src/rules/__tests__/test-utils.ts',
+      ],
+      coveragePathIgnorePatterns: ['/node_modules/'],
+    },
+    {
+      displayName: 'lint',
+      runner: 'jest-runner-eslint',
+      testMatch: ['<rootDir>/**/*.{js,ts}'],
+      testPathIgnorePatterns: ['<rootDir>/lib/.*'],
+      coveragePathIgnorePatterns: ['/node_modules/'],
+    },
+  ],
+} satisfies Config.InitialOptions;
```

Comment:
```suggestion
} satisfies Config;
```

why `satisfies` instead of assigning?

Reply:
because assigning means it _is_ `Config` - so all properties marked as required are such, and optional ones are that outside of the assignment; so TypeScript won't let us do things on those properties later down without existence checks - sadly it's not perfect: we've got to define properties we want to use since the type information is not retained after the assignment, which is why we've got `coveragePathIgnorePatterns` explicitly defined with its default values.

Having said that, I've not checked out `Config.InitialOptions` so that might change things

Reply:
Ok so `jest.Config` is just a re-export of `Config.InitialOptions` - so yeah, we could do either or; I think using `satisfies` is slightly nicer though.

To make it a bit clearer:

<img width="541" alt="image" src="https://github.com/jest-community/eslint-plugin-jest/assets/3151613/a1d0a45f-1b21-4b63-b4c8-e484f8885185">

that's what happens if you use assignment, and it's because to TypeScript `config` is type `Config` rather than the actual object we created.